src/Bridge/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaRegexRestriction.php

```diff
@@ -29,7 +29,7 @@ class PropertySchemaRegexRestriction implements PropertySchemaRestrictionMetadat
      */
     public function create(Constraint $constraint, PropertyMetadata $propertyMetadata): array
     {
-        return $constraint instanceof Regex && $constraint->getHtmlPattern() ? ['pattern' => $constraint->getHtmlPattern()] : [];
+        return $constraint instanceof Regex && $constraint->getHtmlPattern() ? ['pattern' => '^(?:'.$constraint->getHtmlPattern().')$'] : [];
```

Comment:
note: taken directly from https://html.spec.whatwg.org/multipage/input.html#the-pattern-attribute

Reply:
Wait, [apparently](https://json-schema.org/understanding-json-schema/reference/regular_expressions.html) the `(?:...)` syntax (non-capturing group) is not safe for `pattern` 😭 (incidentally, Symfony is already generating simple [capturing] groups, e.g. `/A|B/` gives `.*(A|B).*`)... I'm working on a follow-up PR